### DISCUSSION TITLE
question about Intermediate shaft torque

### DISCUSSION THREAD

**Discussion by AzureJi** [OpenFAST/openfast#2159] [Q&A]
Is there a method in fast test18 that can directly derive the torque of the intermediate shaft, or it needs to be calculated from other output values？And if we need to calculate it, how do you calculate it with those quantities

**Comment by jjonkman** [comment]
Dear @AzureJi,

The ElastoDyn module of OpenFAST does not directly consider an intermediate shaft.  That said, even the high-speed shaft in ElastoDyn is modeled simply--that is, the high-speed shaft speed is modeled simply by multiplying the low-speed shaft speed (on the generator-side of the drivetrain spring/damper) by the gearbox ratio (`GBRatio`).

Without consideration of generator friction (`GBoxEff` = 100%), the torque in an intermediate shaft would equal the low-speed shaft torque divided by the gearbox ratio between the intermediate and low-speed shaft.

The following topic on our forum is likely of interest to you: https://forums.nrel.gov/t/resistant-moment-of-the-rotor-and-of-the-electric-generator/408.

Best regards,

**Comment by AzureJi** [comment]
I've already solved that problem. Thank you very much
